workbench-core/authentication/src/plugins/cognitoUserManagementPlugin.ts

```diff
@@ -67,16 +67,17 @@ export class CognitoUserManagementPlugin implements UserManagementPlugin {
       };
     } catch (error) {
       if (error.name === 'InternalErrorException') {
-        throw new IdpUnavailableError('Cognito encountered an internal error');
+        throw new IdpUnavailableError(error.message);
       }
-      if (error.name === 'NotAuthorizedException') {
-        throw new PluginConfigurationError('Plugin is not authorized to get user info');
-      }
-      if (error.name === 'ResourceNotFoundException') {
-        throw new PluginConfigurationError('Invalid user pool id');
+      if (
+        error.name === 'AccessDeniedException' ||
+        error.name === 'NotAuthorizedException' ||
+        error.name === 'ResourceNotFoundException'
+      ) {
+        throw new PluginConfigurationError(error.message);
```

Comment:
should we be displaying the exact message from cognito?

Reply:
The idea is that the person consuming the package should know why errors are occurring.